Libraries/LibGUI/TextEditor.cpp

```diff
@@ -947,6 +947,9 @@ int TextEditor::content_x_for_position(const TextPosition& position) const
     int x_offset = 0;
     switch (m_text_alignment) {
     case Gfx::TextAlignment::CenterLeft:
+        if (!is_visual_data_up_to_date()) {
+            return m_horizontal_content_padding + x_offset;
+        }
```

Comment:
This is a hack fix that doesn't get at the root of the problem and instead introduces different behavior in a certain state.

Based on what @FalseHonesty is saying, it sounds like the problem is that we're trying to set the cursor position while the visual line data is invalid. I think a proper fix would be to prevent `set_cursor()` from trying to use the visual lines for anything while they are invalid.

I think we can just skip the whole `if (m_cursor != position) {` block in `set_cursor()` if there's a pending reflow. Does that make sense?